### PR TITLE
Feature 5 html layout editor screen

### PR DESCRIPTION
**Pull request by Jdallos**
-Needs merging with any changes to develop branch from feature 3 branch after 30/03/23

### CODE REVIEW

**Review by EVP1989 — COMMENTED** [review]
Approved

File: src/components/screens/layout-screen-html.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  layouts: PageLayout[];
+  subLayouts: SubLayout[];
+  setLayouts: typeof setLayouts;
+  setSelectedLayout: typeof setSelectedLayout;
+}
+
+/**
+ * Component for html layout screen
+ */
+export const LayoutScreenHTML: React.FC<Props> = ({
```
Comment (villejuutila) [inline src/components/screens/layout-screen-html.tsx:49]:
```suggestion
const LayoutScreenHTML: React.FC<Props> = ({
```

File: src/components/screens/layout-screen-html.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+  history,
+  keycloak,
+  deviceModels,
+  layout,
+  classes
+}) => {
+  const [ name, setName ] = useState("");
+  const [ deviceModelId, setDeviceModelId ] = useState("");
+  const [ screenOrientation, setScreenOrientation ] = useState<ScreenOrientation>(ScreenOrientation.Portrait);
+  const [ view, setView ] = useState("VISUAL")
```
Comment (villejuutila) [inline src/components/screens/layout-screen-html.tsx:59]:
```suggestion
  const [ view, setView ] = useState<"VISUAL" | "CODE">("VISUAL")
```

File: src/components/screens/layout-screen-html.tsx [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+  const [ dataChanged, setDataChanged ] = useState(false);
+
```
Comment (villejuutila) [inline src/components/screens/layout-screen-html.tsx:61]: Add functionality for fetching the layout from the API.
Currently if you reload this page, the selected layout is lost from state and this screen is left in an endless-loop.
Also navigate back to listing if the layout couldn't be found.

File: src/components/screens/layout-screen-html.tsx [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+  const getActionButtons = (): ActionButton[] => (
+    [
+      {
+        name: view === "CODE" ?
+          strings.exhibitionLayouts.editView.switchToVisualButton :
+          strings.exhibitionLayouts.editView.switchToCodeButton,
+        action: () => view === "CODE" ? setView("VISUAL") : setView("CODE"),
+      },
+      {
+        name: strings.exhibitionLayouts.editView.saveButton,
```
Comment (villejuutila) [inline src/components/screens/layout-screen-html.tsx:106]: Save functionality is in place in the API so therefore this action could just save it.

**Review by villejuutila — CHANGES_REQUESTED** [review]

File: src/components/screens/layout-screen-html.tsx (hunk 2 above)
Comment (villejuutila) [reply src/components/screens/layout-screen-html.tsx:59]: Or enumerate this as something like LAYOUT_EDITOR_VIEW, whichever seems best option.

Comment (Jdallos) [reply src/components/screens/layout-screen-html.tsx:59]: Set as enum

**Review by villejuutila — APPROVED** [review]

**Pull request merged**